fs_gcsfs/_gcsfs.py

```diff
@@ -186,14 +186,12 @@ def _scandir(self, path: str, return_info: bool = False, namespaces: List[str] =
         dir_key = self._path_to_dir_key(_path)
 
         if dir_key == "/":
-            # In case we want to list the root directory, no prefix is necessary
-            prefix = ""
-        else:
-            prefix = dir_key
-        prefix_len = len(prefix)
+            # In case we want to list the bucket, dir_key needs to be the empty string
+            dir_key = ""
+        prefix_len = len(dir_key)
```

Comment:
I do not really understand what the problem was and why it is fixed now.

As far as I can see you only mutated `dir_key` instead of introducing a new variable `prefix` here (which I found to be more explicit, as the `dir_key` is actually correct, it just does not work with `list_blobs`).

Reply:
The problem was that `prefix` was overwritten when iterating over `prefixes` and that some calls below used `dir_key` instead of `prefix`.

Reply:
I think it would be beneficial, if the behaviour of `_path_to_key(root)` and `_path_to_dir_key(root)` would be clearly defined if the root_path of the file system is the root of the bucket and `root` is any valid pyfilesystem specifier for the root of a filesystem.